shard.go

```diff
@@ -10,19 +10,25 @@ import (
 
 type onRemoveCallback func(wrappedEntry []byte, reason RemoveReason)
 
+type metadata struct {
```

Comment:
Why unexported? Same question about single field.

Reply:
I followed the same logic as `cacheShard` but I can change that if you want.

Reply:
That's a big difference. We're exposing `metadata` type to the user (from `BigCache` method, and exposing unexported type is a strange thing.

Reply:
should be fixed now

Reply:
Please, expose `requestCount` too (should be `RequestCount`) 😉

Reply:
done :+1: